### PR TITLE
Hoist Statics from DecoratedComponent

### PR DESCRIPTION
**Pull request by umidbekkarimov**
Fix for #38 

I had problems while running `build` locally so I updated and add all dev dependencies to `package.json`. Also I slightly fixed `build` script.

### CODE REVIEW

**Comment by kof** [comment]
I am not sure copying static props to the HOC is a good thing. Need more input for pros and cons.

**Comment by umidbekkarimov** [comment]
@kof 

> I am not sure copying static props to the HOC is a good thing.

Some powerful libs that provide HOC and use [hoist-non-react-statics](https://github.com/mridgway/hoist-non-react-statics) in it:

* react-redux - [@connect](https://github.com/reactjs/react-redux/blob/master/src/components/connect.js#L365) 
* react-router - [@withRouter](https://github.com/reactjs/react-router/blob/master/modules/withRouter.js#L20)
* redux-form - [@reduxForm](https://github.com/erikras/redux-form/blob/master/src/createReduxForm.js#L39)

>  Need more input for pros and cons.

I have `SelectField` component with it's static method `SelectField.getOptionKey` which is used like simple helper to obtain value by key.

I used it in this way:

```javascript

import SelectField from './SelectField';

function PackageSelectField({
    packagePrices,

    fields: {
        packageItem
    }
}) {

    return (
        <SelectField
            {...packageItem}
            options={packagePrices}
            hintText="Choose Package"

            getOptionText={SelectField.getOptionKey('text')}
            getOptionValue={SelectField.getOptionKey('value')}
        />
    );
}

```

File: package.json [hunk 1]
```diff
@@ -4,9 +4,9 @@
   "description": "JSS mixin for React components",
   "main": "lib/index.js",
   "scripts": {
-    "build": "babel src --out-dir lib",
+    "build": "./node_modules/.bin/cross-env NODE_ENV=production BABEL_ENV=commonjs ./node_modules/.bin/babel src --out-dir lib",
```
Comment (kof) [inline package.json]: full path should not be needed, why? npm adds it automatically ...

Comment (umidbekkarimov) [inline package.json]: Really didn't know that, I had problem with it in `travis-ci` once so i add full path every time.

**Comment by kof** [comment]
looks good

**Comment by kof** [comment]
don't forget to lint

**Comment by kof** [comment]
huh, after updating dependencies linter stopped having that weird error.

**Comment by umidbekkarimov** [comment]
@kof yes, i updated `eslint` version because of strange errors

**Comment by kof** [comment]
published

**Pull request merged**